[T7][W09-B3]

e0003823 opened the pull request:
V0.0

nnamon committed:
Changed the name of the project.

nnamon committed:
Replaced the project name in README.md with Suru.

nnamon committed:
Modified README to show Travis CI build status for this repository.

nnamon committed:
Modified README to show Travis CI build status for this repository - fix

shawnljj committed:
Updated User Guide

shawnljj committed:
Updated DevGuide.md

nnamon committed:
Update the Developer Guide with the latest product survey.

nnamon committed:
Updated the User Guide.

ephyra committed:
Update DeveloperGuide with work distribution comment.

shawnljj committed:
Updated User Guide, mainly Commands Summary

shawnljj committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

# Conflicts:
#	docs/DeveloperGuide.md

ephyra committed:
update name in .project

shawnljj committed:
additional commit to test travis

Commit:
Update Coverage Status

Commit:
Update Appveyor Badge

Commit:
Update Codacy Badge

ephyra committed:
refactor addressBook to TaskManager, not completed

shawnljj committed:
First run of About Us page.

shawnljj committed:
Designed mockup of UI and added Ui.jpg to the User guide.

shawnljj committed:
Revert "Designed mockup of UI and added Ui.jpg to the User guide."

This reverts commit 9b7a90f77dc8fcca6f4d3a53852b825caa96757b.

e0003823 committed:
Re-commit User Guide after reverting previous one due to commit performed by wrong account.

e0003823 committed:
Updated Mockup image for Readme

ephyra committed:
refactor addressBook to TaskManager, work in progress

ephyra committed:
refactor addressBook to taskManager, Main folder done

e0003823 committed:
removed background from favicon

Commit:
Updated Readme.MD to Suru context

ephyra committed:
refactor StartDate and EndDate to StartTime and EndTime and also change relevant regex for validation.

ephyra committed:
Change StartTime and EndTime validation regex to accept empty string

e0003823 committed:
Update DeveloperGuide.md

Add redo user story and use case

e0003823 committed:
Merge branch 'v0.1-AboutUs-updates'

# Conflicts:
#	docs/AboutUs.md

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

e0003823 committed:
Update AboutUs.md

Moved documentation responsibilities to Mustaqiim

Commit:
Removed trailing whitespaces for Developer & User Guide

Commit:
Removed whitespace

hwkchia commented on docs/UserGuide.md:
```diff
@@ -11,25 +11,22 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-1. Download the latest `addressbook.jar` from the [releases](../../../releases) tab.
-2. Copy the file to the folder you want to use as the home folder for your Address Book.
-3. Double-click the file to start the app. The GUI should appear in a few seconds.
-   > <img src="images/Ui.png" width="600">
```
Add some vertical space before the mockup image? By the way, nice mockup. 👍

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+
+Adds a task with a due date to the task manager
+
+Format: `add DESCRIPTION by DUE_DATE`
```
👍  Good, you made an effort not to use command switches or flags. In the meantime, you should also consider the case where `by` might also make up part of the description.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+
+Examples:
+
+* `add write essay by 13/7/17`
```
Are date/time formats also flexible?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+
+Examples:
+
+* `add write essay from 13/7/17`
```
May I suggest a different task description, just to give some variety.  `write essay from` also sounds weird... or maybe it's just me. 😛

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+Shows a list of all tasks in the task manager.
 
-Shows a list of all persons in the address book.<br>
 Format: `list`
```
I guess you can skip this since there is no another format.  Likewise for `list done` and `list undone`.  Or maybe combine all three under `list` so that now you can show the different formats?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+  Edits the description of the task to `buy groceries` and due date to `13/5/17` respectively.
+
+* `edit 2 buy groceries`
+  Edits the description of the 2nd task to be `buy groceries`.
```
👍  This is good... the more flexible it is, the better.  Of course, you will need to put in a lot of work in the parser.  I assure you that if you see to it, then the payoff will be very high. 😄

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+  Deletes the 2nd task in the list.
+* `find Essay`
+  `delete 1`
+  Deletes the 1st task in the results of the `find` command.
```
It seems that your intention is to have the `list` and `find` commands separate from `delete` command? But form the way it's rendered, it looks like on entire command, e.g. `find Essay delete 1`.

Anyway, if indeed it's supposed to be separate, then kudos to you. 👍

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+> * All properties are searched e.g. `DUE_DATE`, `TAGS`, `DESCRIPTION` etc.
+> * Only full words will be matched e.g. `Ess` will not match `Essay`
+> * Tasks matching at least one keyword will be returned (i.e. `OR` search).
+    e.g. `Essay` will match `Essay Writing Competition`
```
Performing a partial search would be good too.  When it comes to search, no one does it better than google. 😄

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-Format: `delete INDEX`
+Finds undone tasks whose properties contain any of the given keywords.
+
+Format: `find undone KEYWORD [MORE_KEYWORDS]`
```
Yes, you can certainly find within a particular scope.  Consider finding free slots within a time period also. This is very useful.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+
+Format: `undo`
+
+> * Only reverses operations that manipulate data, i.e. `add`, `edit` and `delete`
```
Is that all?  How about `check`, `uncheck`, ... ?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+  `uncheck 2`
+  Unchecks the 2nd task in the listing marking it as undone.
+* `find Essay`
+  `uncheck 1`
```
Rendering problem here also?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+  `check 2`
+  Checks off the 2nd task in the listing as done.
+* `find Essay`
+  `check 1`
```
Rendering problem?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-Selects the person identified by the index number used in the last person listing.<br>
-Format: `select INDEX`
+### 2.20. See usage instructions: `help`
+Shows a list of commands that Suru accepts.
```
Wans't this mentioned in 2.2?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+
+* `loadfrom` `FILE_PATH`
+
+### 2.24. Add reminders to task: `add DESCRIPTION reminders: ...`
```
I guess from here on are your lower priority features?  It would be good to include them in the relevant sections as and when you make time for them.

hwkchia reviewed (commented):
A very good job in illustrating the usage of your product.  Well done and keep up the work rate!

You can refer to more guidance here. 
https://drive.google.com/open?id=1kAth-pTOzwCJFVjS6KU8O50bCvijDqpb6GsOyJTD0Ds

hwkchia commented:
User Guide has been reviewed.  The DevGuide will be reviewed soon, so do not close the PR yet.

e0003823 committed:
Changed mockup image

e0003823 committed:
Improved readability for mockup

nnamon committed:
Update AboutUs.md and add nnamon image

nnamon committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-W09-B3/main

ephyra committed:
refactor xmlSerializableAddressBook to xmlSerializableTaskManager, edit tag prefix from "t/" to "#"

ephyra committed:
refactor xmlSerializableAddressBook to xmlSerializableTaskManager, edit tag prefix from "t/" to "#" part2

e0003823 committed:
Update UserGuide.md

Commit:
TestDebugging Part 1

e0003823 committed:
Modify asserts that check for valid time

Commit:
Test Refactor Done

e0003823 committed:
updated validFile.fxml and Fixed last failure

Commit:
Merge branch 'v0.1-refactoring-mus'

Commit:
Added EOL

e0003823 committed:
Test check style

e0003823 committed:
Checkstyle formatting fixes

ephyra committed:
Change data-type of CompletionStatus, remove validation of CompletionStatus, add swapStatus and edit parser and logic to work with above changes

ephyra committed:
fix AddCommandParser in accordance to checkstyle

Commit:
Checkoff Task Feature V1

ephyra committed:
test commit

ephyra committed:
no message

Commit:
Merge branch 'v0.2-data-type-and-parser-changes' of https://github.com/CS2103JAN2017-W09-B3/main into v0.2-feature-checkofftask

ephyra committed:
change sample data

Commit:
Merge branch 'v0.2-data-type-and-parser-changes' of https://github.com/CS2103JAN2017-W09-B3/main into v0.2-feature-checkofftask

ephyra committed:
Reverse last 2 commits

Commit:
Debug Test Cases

ephyra committed:
Fix revert commits.

Commit:
Merge branch 'v0.2-data-type-and-parser-changes' of https://github.com/CS2103JAN2017-W09-B3/main into v0.2-feature-checkofftask

# Conflicts:
#	src/test/java/guitests/EditCommandTest.java
#	src/test/java/guitests/guihandles/PersonListPanelHandle.java

Commit:
Resolve Merge Conflicts

Commit:
Fixed Checkstyle Issues

e0003823 committed:
Modified UI for Layout to match mockup

e0003823 committed:
Commit new files

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main into v0.2-UI-Redesign

# Conflicts:
#	src/test/java/guitests/guihandles/PersonCardHandle.java
#	src/test/java/seedu/task/model/person/AddressTest.java
#	src/test/java/seedu/task/testutil/PersonBuilder.java
#	src/test/java/seedu/task/testutil/TestPerson.java
#	src/test/java/seedu/task/testutil/TestUtil.java

e0003823 committed:
Set theme jekyll-theme-slate

e0003823 committed:
Set theme jekyll-theme-leap-day

e0003823 committed:
Change link for Help

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

e0003823 committed:
Add newline at EOF

e0003823 committed:
Add new line to file

e0003823 committed:
Checkstyle fixes

e0003823 committed:
Checkstyle fixes again

e0003823 committed:
Add user’s commands to chat

Commit:
Abstracted Task Completed and created two child classes Check & Unchecked. Included checks to see if task has already been checked or unchecked

Commit:
Merge branch 'master' of https://github.com/CS2103JAN2017-W09-B3/main

Commit:
Checkstyle changes!

e0003823 committed:
Update ContactUs.md

Changed email address

e0003823 committed:
Update ContactUs.md

Updated issue tracker link

e0003823 committed:
Update UserGuide.md

Added introduction, fixed anchor links, improved language to be more personal, standardized formatting for all features

e0003823 committed:
Add files via upload

Added images for user guide

e0003823 committed:
Update UserGuide.md

Changed contact email

e0003823 committed:
Update UserGuide.md

Repositioned UI image

e0003823 committed:
Update DeveloperGuide.md

Intermediate commit for Developer Guide

e0003823 committed:
Set theme jekyll-theme-leap-day

e0003823 committed:
Update DeveloperGuide.md

Changed markdown comment to HTML comment to support GitHub Pages

e0003823 committed:
Set theme jekyll-theme-slate

e0003823 committed:
Set theme jekyll-theme-cayman

Commit:
V 0.3.1 undo implementation only for Add command

e0003823 committed:
Update DeveloperGuide.md

e0003823 committed:
Update UserGuide.md

e0003823 committed:
Update DeveloperGuide.md

HTML comment style

Commit:
Undo working for add, edit & delete

e0003823 committed:
Moved ChatList.java model to the correct model package

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

e0003823 committed:
Update UserGuide.md

fixed find.jpg image url

e0003823 committed:
Update UserGuide.md

Modified tables

e0003823 committed:
Tabulated Command Summary

e0003823 committed:
Made improvements based on feedback

ephyra committed:
Add list checked and list unchecked commands

Commit:
Redesign of undo function

Commit:
Undo bug fixes

e0003823 committed:
Styled messages with sender and receiver bubbles.

Changed resultDisplay textArea to a text

Adjusted sizes so that chat pane takes up most of the height of the right pane.

e0003823 committed:
Changed App Title

e0003823 committed:
Merge pull request #63 from CS2103JAN2017-W09-B3/v0.3-list-checked-unchecked

Add list checked and list unchecked commands

Commit:
Merge branch 'master' of https://github.com/CS2103JAN2017-W09-B3/main into v0.3-undo

# Conflicts:
#	src/main/java/seedu/task/logic/parser/Parser.java

Commit:
Resolve Conflict

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

Commit:
Checkstyle Changes

Commit:
Morezzzz Checkstylezzz Changess!

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

nnamon committed:
Modify build.gradle with the actual application name to fix the run build case.

e0003823 committed:
Merge branch 'master' of github.com:CS2103JAN2017-W09-B3/main

e0003823 committed:
Fix check style

Commit:
Fixed Undo Edit bug

nnamon committed:
Modified build.gradle to include the natty dependency in gradle syncing

nnamon committed:
Modified build.gradle to include the natty dependency in gradle syncing

e0003823 committed:
Updated Sequence Diagrams and Ui Diagram.

nnamon committed:
Add logging to Parser. Created a summary of commands. Created test class for the Parser class.

nnamon committed:
Add logging to Parser. Created a summary of commands. Created test class for the Parser class.

nnamon committed:
Merge branch 'v0.4-review-commands' of https://github.com/CS2103JAN2017-W09-B3/main into v0.4-review-commands

nnamon committed:
Modify the UndoManager to accept COMMAND_WORDs from the respective Command classes.

nnamon committed:
Merge pull request #68 from CS2103JAN2017-W09-B3/v0.4-review-commands

V0.4 review commands

nnamon committed:
Modify the Add command to match arguments using regex.
Add tests for the Parser.
Implemented tests for the Add command.
Added a Natty Date Utility class.
Add tests for the Natty Date Utility class.
Added a document of Command Summaries.

nnamon committed:
Add support for tags in the Add command.

nnamon committed:
Change StartTime to accept a Date object in its constructor. Change AddCommand to use Date objects. Fixed all associated tests.

nnamon committed:
Change EndTime to accept a Date object in its constructor. Change AddCommand to use Date objects. Fixed all associated tests.

ephyra committed:
update model component class diagram

nnamon committed:
Update the parser tests. All AddCommand tests are running and are functional as of this commit.

nnamon committed:
Modify EditCommandParser to parse the new commands. Added the corresponding tests.

nnamon committed:
Modify UserGuide and DeveloperGuide to use US format dates.

nnamon committed:
Merge pull request #71 from CS2103JAN2017-W09-B3/v0.4-review-commands

V0.4 review commands

nnamon committed:
Made the parser accept case insensitive commands.

nnamon committed:
Setup the skeleton to handle LoadFrom, SaveTo, and Redo

nnamon committed:
Merge pull request #74 from CS2103JAN2017-W09-B3/v0.4-review-commands

V0.4 review commands

nnamon committed:
Modified the way StartTime and EndTime converts to string by default. Fixed the command generator for the LogicManagerTests. Removed one obsolete test.

nnamon committed:
Merge pull request #76 from CS2103JAN2017-W09-B3/v0.4-fixTravis

Modified the way StartTime and EndTime converts to string by default.…

nnamon committed:
Fix checkstyle

nnamon committed:
Merge pull request #77 from CS2103JAN2017-W09-B3/v0.4-fixTravis

Fix checkstyle

nnamon committed:
Fix the ConfigTest

nnamon committed:
Merge pull request #78 from CS2103JAN2017-W09-B3/v0.4-fixTravis

Fix the ConfigTest

nnamon committed:
Fix checkstyle warning for TestUtil class

nnamon committed:
Clean up Tests. Refactor old addressbook code to reflect a task manager.

e0003823 committed:
Updated Diagrams and replaced images in Dev Guide

nnamon committed:
Modify LogicClassDiagram

nnamon committed:
Copy LogicClassDiagram.png to images

nnamon committed:
Modify and update images for the LogicClassDiagram

nnamon committed:
Merge branch 'master' into v0.3-Architecture-Diagrams

nnamon committed:
Merge pull request #79 from CS2103JAN2017-W09-B3/v0.3-Architecture-Diagrams

V0.3 architecture diagrams
